src/vis/Vis.tsx

```diff
@@ -69,54 +74,85 @@ export function Vis({
   externalConfig?: IVisConfig;
   hideSidebar?: boolean;
 }) {
-  const [visConfig, setVisConfig] = React.useState<IVisConfig>(
-    externalConfig || columns.filter((c) => c.type === EColumnTypes.NUMERICAL).length > 1
+  // Each time you switch between vis config types, there is one render where the config is inconsistent with the type before the merge functions in the useEffect below can be called.
+  // To ensure that we never render an incosistent config, keep a consistent and a current in the config. Always render the consistent.
+  // TODO:: probably change the names, especially "current" and "inconsistentVisConfig" to something more useful.
```

Comment:
From the state properties, I would either expect something like `consistent` + `inconsistent` or `current` + `previous`. If they are matching you could also name your local variables `visConfig` and `inconsistentVisConfig`.

In general, however, we should try to solve the root of the inconsistent vis config. @puehringer do you have an idea if there is still some issue with our vis config setup?

Reply:
We agreed in a chat that we will keep the solution for now as it is and might refactor it later with issue https://github.com/datavisyn/reprovisyn/issues/304.